pkg/common/report/weather_report.go

```diff
@@ -0,0 +1,135 @@
+package report
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/openshift/osde2e/pkg/common/aws"
+	"github.com/openshift/osde2e/pkg/common/templates"
+)
+
+var markdownTemplate *template.Template
+
+func init() {
+	var err error
+
+	markdownTemplate, err = templates.LoadTemplate("/assets/reports/markdown.template")
+
+	if err != nil {
+		panic(fmt.Sprintf("error loading markdown template: %v", err))
+	}
+}
+
+// WeatherReport is the weather report.
+type WeatherReport struct {
+	ReportDate time.Time   `json:"reportDate"`
+	Jobs       []JobReport `json:"jobs"`
+
+	// We want the sort interface so that we can sort jobs and produce stable, comparable reports.
+	sort.Interface `json:"-"`
+}
+
+// JobReport is a report for an individual job.
+type JobReport struct {
+	Name         string   `json:"name"`
+	Viable       bool     `json:"viable"`
+	Versions     []string `json:"versions"`
+	FailingTests []string `json:"failingTests,omitempty"`
+}
+
+// Len is the number of jobs in the weather report.
+func (w WeatherReport) Len() int {
+	return len(w.Jobs)
+}
+
+// Less reports whether the element with index i should sort before the element with index j.
```

Comment:
Is this used anywhere?

Reply:
Yeah, these are implicitly used by sort -- these are part of the sorting Interface.